text/0517-io-os-reform.md

```diff
@@ -556,6 +556,15 @@ buffers as input. This has multiple benefits:
   results are wanted, one can use `read_to_end` and convert to a
   `String` only at the end.
 
+`read_to_end` and `read_to_string` return a `Result` with either the
+number of bytes read or an error.  If a short read occurs, that call
+gives the (short) data and returns the number of bytes written into
+that buffer.  If the error was non-transient, future calls shall return
+the reason for the error (usually `EndOfFile`).
```

Comment:
These methods worked perfectly fine in old_io. There is no reason to remove them.